services/web/server/tests/unit/test_activity.py

```diff
@@ -0,0 +1,118 @@
+from utils_assert import assert_status
+
+
+def future_with_result(result):
+    f = Future()
+    f.set_result(result)
+    return f
+
+
+@pytest.fixture
+def mocked_login_required(mocker):
+    mock = mocker.patch(
+        'simcore_service_webserver.login.decorators.login_required',
+        lambda h: h)
+    importlib.reload(handlers)
+    return mock
+
+@pytest.fixture
+def mocked_monitoring(loop, mocker, activity_data):
+    prometheus_data = activity_data.get('prometheus')
+    cpu_ret = prometheus_data.get('cpu_return')
+    mocker.patch('simcore_service_webserver.activity.handlers.get_cpu_usage',
+        return_value=future_with_result(cpu_ret))
+
+    mem_ret = prometheus_data.get('memory_return')
+    mocker.patch('simcore_service_webserver.activity.handlers.get_memory_usage',
+        return_value=future_with_result(mem_ret))
+
+    labels_ret = prometheus_data.get('labels_return')
+    mocker.patch('simcore_service_webserver.activity.handlers.get_container_metric_for_labels',
+        return_value=future_with_result(labels_ret))
+
+    celery_data = activity_data.get('celery')
+    celery_ret = celery_data.get('celery_return')
+    mocker.patch('simcore_service_webserver.activity.handlers.get_celery_reserved',
+        return_value=future_with_result(celery_ret))
+
+@pytest.fixture
+def mocked_monitoring_down(mocker):
+    mocker.patch(
+        'simcore_service_webserver.activity.handlers.query_prometheus',
+        side_effect=ClientConnectionError)
+    mocker.patch(
+        'simcore_service_webserver.activity.handlers.celery_reserved',
+        side_effect=ClientConnectionError)
+    return mocker
+
+@pytest.fixture
+def app_config(fake_data_dir: Path, osparc_simcore_root_dir: Path):
+    with open(fake_data_dir/"test_activity_config.yml") as fh:
+        content = fh.read()
+        config = content.replace("${OSPARC_SIMCORE_REPO_ROOTDIR}", str(osparc_simcore_root_dir))
+
+    return yaml.load(config)
+
+@pytest.fixture
+def client(loop, aiohttp_client, app_config):
+    app = create_safe_application(app_config)
+
+    setup_session(app)
+    setup_security(app)
+    setup_rest(app)
+    setup_activity(app)
+
+    cli = loop.run_until_complete(aiohttp_client(app))
+    return cli
+
+
+async def test_has_login_required(client):
+    resp = await client.get('/v0/activity/status')
+    _data, _error = await assert_status(resp, web.HTTPUnauthorized)
+
+async def test_monitoring_up(mocked_login_required, mocked_monitoring, client):
+    resp = await client.get('/v0/activity/status')
+    data, _ = await assert_status(resp, web.HTTPOk)
+    assert '35f95ad4-67b8-4ed8-bd55-84a5d600e687' in data, 'Queued node not present'
+    assert '894dd8d5-de3b-4767-950c-7c3ed8f51d8c' in data, 'Queued node not present'
+    celery = data.get('35f95ad4-67b8-4ed8-bd55-84a5d600e687')
+    prometheus = data.get('894dd8d5-de3b-4767-950c-7c3ed8f51d8c')
+    assert 'queued' in celery, 'There is no queued key for queued node'
```

Comment:
MINOR: it would be a more effective test if instead of doing the structural/values check by hand you would use a jsonschema validator and check against the data schema

Reply:
certainly... also the values you check them with a JSON schema?

Reply:
no, you validate the structure of your data against your schema. Values will raise an error ONLY if they do not fit e.g. a pattern, a type, an enum, etc ... but NOT the actual data